content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+
+- Interface
+
+- Non-Access Modifier keyword “Abstract” Implementation
 
-- 1: Non-Access Modifier keyword “Abstract” Implementation
 The abstract access modifier keyword can only be used with classes, methods but not with variables. Methods with the keyword abstract are methods without implementation.
 
-To implement an abstract class in Java, use the abstract keyword immediately before the class keyword preceding the class name e.g public abstract class Circle, where Circle is the class name. Use this pattern for abstract method implementation, public abstract returnType methodName() i.e public abstract double getArea().
+To implement an abstract class in Java, use the abstract keyword e.g public abstract class Circle, where Circle is the class name. 
+Use this pattern for abstract method implementation, public abstract returnType methodName() i.e public abstract double getArea().
 
 Code snippet
+
 ```java
+
 public abstract class Dog {
-	
-	public abstract void run();
 
+   public abstract void run();
 
 }
+
 ```
 
-With the above code snippet, we've created an abstract class and an abstract method but one pertinent feature of abstract classes is that objects cannot be instantiated or created from it.But don’t forget, classes only exist for the purpose of creating an object. To be able to create an object from an abstract class we have to create a new class that inherit from the superclass(base, parent) class by using the extends keyword.
+With the above code snippet, we've created an abstract class and an abstract method but one pertinent feature of abstract class is that objects cannot be instantiated or created from it. But don’t forget, classes only exist to create an object. To be able to create an object from an abstract class. We have to create a new class that inherits from the superclass(base, parent) class by using the extends keyword.
 
 Code snippet
+
 ```java
+
 public class MainDog extends Dog {
 
-    //abstract method body definition
-	public void run(){
-		//in this class the method is implemented
-	    System.out.println(“Dog is running”);
-    }
-	
-	public static void main(Strings [] args){
-		
-		MainDog dog = new MainDog();
+   //abstract method body definition
 
-		//abstract method invocation
-		dog.run();
-    }
+   public void run(){
+
+       //in this class the method is implemented
+
+       System.out.println(“Dog is running”);
+
+   }
+
+   public static void main(Strings [] args){
+
+       MainDog dog = new MainDog();
+
+       //abstract method invocation
+
+       dog.run();
+
+   }
 
 }
+
 ```
 
--[2]: Interface implementation
+- Interface implementation
+
 An interface contains public, abstract method signatures (methods without implementation). It is also important to note that interfaces have no constructor and may contain final and static variables. Abstraction implementation by the interface method is made possible by the use of the Implements keyword.
```

Comment:
> Abstraction implementation by the interface method is made possible by the use of the Implements keyword.

don't capitalize the word **implements** to keep it looking like Java code and also because it is not a [proper noun](https://www.grammarly.com/blog/proper-nouns/). Also, surround it in backtick:

> Abstraction implementation by the interface method is made possible by the use of the `implements` keyword.